whisper.go

```diff
@@ -127,22 +150,47 @@ func ParseRetentionDefs(retentionDefs string) (Retentions, error) {
 	return retentions, nil
 }
 
+type file interface {
+	Seek(offset int64, whence int) (ret int64, err error)
+	Fd() uintptr
+	ReadAt(b []byte, off int64) (n int, err error)
+	WriteAt(b []byte, off int64) (n int, err error)
+	Read(b []byte) (n int, err error)
+	Name() string
+	Close() error
+	Write(b []byte) (n int, err error)
+}
+
 /*
 	Represents a Whisper database file.
 */
 type Whisper struct {
-	file *os.File
+	// file *os.File
+	file file
 
 	// Metadata
 	aggregationMethod AggregationMethod
 	maxRetention      int
 	xFilesFactor      float32
 	archives          []*archiveInfo
+
+	compressed             bool
+	compVersion            uint8
+	pointsPerBlock         int
+	avgCompressedPointSize float32
+
+	crc32 uint32
+
+	opts     *Options
+	Extended bool
 }
 
 // Wrappers for whisper.file operations
 func (whisper *Whisper) fileWriteAt(b []byte, off int64) error {
 	_, err := whisper.file.WriteAt(b, off)
+	if err != nil {
+		panic(err)
```

Comment:
What's the point to panic here and then return 'err'?

For libraries it's usually better to return error if any.

Reply:
Thanks for pointing it out, that's just debug code. Removed.